rust_src/src/marker.rs

```diff
@@ -559,7 +559,7 @@ impl LispBufferRef {
 
 /// Return the byte position corresponding to CHARPOS in B.
 #[no_mangle]
-pub extern "C" fn buf_charpos_to_bytepos(b: *mut Lisp_Buffer, charpos: isize) -> isize {
+pub unsafe extern "C" fn buf_charpos_to_bytepos(b: *mut Lisp_Buffer, charpos: isize) -> isize {
```

Comment:
Why does clippy recommend making this function unsafe?

Reply:
I'm not sure exactly, but I think it requires that any function that directly manipulates a raw pointer be marked unsafe.

Reply:
@db48x clippy warns whenever there is `unsafe` within a function and the function signature does not reflect it.

Reply:
That doesn't make any sense. A function that is marked as "unsafe" doesn't need to have any unsafe blocks inside of it; in fact it generally should not.

Reply:
Is it not possible to configure clippy like rustfmt ?

Reply:
It is, but the safety recommendations seem reasonable to me. Or rather, since I don't completely understand the logic behind them, it seems reasonable to me to do whatever Clippy says with respect to safety unless there is a specific reason not to.